File: 250702/XOR 결과 최대 만들기/max-of-xor.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const [n, m] = input[0].split(' ').map(Number);
const a = input[1].split(' ').map(Number);

let max = 0
function dfs(start, count, xor){
    if(count === m){
        max = Math.max(max, xor)
        return;
    }
    for(let i = start; i < n; i++){
        dfs(i + 1, count + 1, xor ^ a[i]);
    }
}
dfs(0, 0, 0)
console.log(max)

File: 260122/XOR 결과 최대 만들기/max-of-xor.py
n, m = map(int, input().split())
A = list(map(int, input().split()))
visited = [False for _ in range(n)]

max_ans = 0

def dfs(start, cnt):
    global max_ans

    if cnt == m:
        temp = 0
        for i in range(n):
            if visited[i]:
                temp ^= A[i]

        max_ans = max(max_ans, temp)
        return
    if start == n:
        return

    dfs(start + 1, cnt)

    visited[start] = True
    dfs(start + 1, cnt + 1)
    visited[start] = False


dfs(0, 0)

print(max_ans)

